sync-core/src/test/java/com/cloudant/sync/datastore/AttachmentTest.java

```diff
@@ -66,6 +62,11 @@ public void setAndGetAttachmentsTest() throws Exception {
             SavedAttachment savedAtt2 = (SavedAttachment) datastore.attachmentsForRevision
                     (newRevision).get(0);
             Assert.assertArrayEquals(expectedSha1, savedAtt2.key);
+
+            //Assert file length and saved attachment length
+            long expectedLength = f.length();
+            Assert.assertEquals(expectedLength,savedAtt.getSize());
+            Assert.assertEquals(expectedLength,savedAtt2.getSize());
```

Comment:
Added test cases in eac97bf8469ddacf68ef93837a45603a67a0313f and 4bacefb14cbfcaaf30e6db9faf57f2f63d6d2e5e